magma/circuit.py

```diff
@@ -264,9 +264,8 @@ def __new__(metacls, name, bases, dct):
         return cls
 
     def __call__(cls, *largs, **kwargs):
-        if get_debug_mode():
-            debug_info = get_callee_frame_info()
-            kwargs["debug_info"] = debug_info
+        debug_info = get_debug_info(3)
+        kwargs["debug_info"] = debug_info
```

Comment:
nit
```suggestion
        kwargs["debug_info"] = get_debug_info(3)
```

(unless this screws up the callstack lol)